client.go

```diff
@@ -428,6 +446,31 @@ func (c *Client) handleMessage(msgType int, data []byte) error {
 				return err
 			}
 			c.sigChainBlockHash = sigChainBlockHash
+
+		// client auth
+		case authChallengeAction:
+			var challenge string
+			if err := json.Unmarshal(*msg["Challenge"], &challenge); err != nil {
+				return err
+			}
+			byteChallenge, _ := hex.DecodeString(challenge)
+
+			var clientSalt [32]byte
+			rand.Reader.Read(clientSalt[:])
+
+			// add client salt to the node's challenge
+			byteChallenge = append(byteChallenge, clientSalt[:]...)
```

Comment:
I would suggest we sign `sha256(byteChallenge)`

Reply:
yes, fixed